include/Basis/SpinorBasis/USpinorBasis.hpp

```diff
@@ -0,0 +1,359 @@
+     * 
+     *  @param component                the spin component
+     * 
+     *  @param U            the unitary transformation matrix that transforms both the alpha- and beta components
+     */
+    void rotate(const TransformationMatrix<ExpansionScalar>& U, const SpinComponent& component) {
+        this->spinor_basis[component].rotate();
+    }
+
+
+    /**
+     *  Rotate the spinor basis to another one using the given unitary transformation matrix
+     * 
+     *  @param U            the unitary transformation matrix that transforms both the alpha- and beta components
+     */
+    void rotate(const TransformationMatrix<ExpansionScalar>& U) {
+        this->rotate(SpinComponent::ALPHA);
+        this->rotate(SpinComponent::BETA);
+    }
+
+    /**
+     *  Rotate the spinor basis of the requested component to another one using the unitary transformation matrix that corresponds to the given Jacobi rotation parameters
+     * 
+     *  @param jacobi_rotation_parameters       the Jacobi rotation parameters (p, q, angle) that are used to specify a Jacobi rotation: we use the (cos, sin, -sin, cos) definition for the Jacobi rotation matrix
+     * 
+     *  @note this function is only available for real spinor bases because Jacobi rotation parameters generate real rotations
+     */
+    template<typename Z = ExpansionScalar>
+    enable_if_t<std::is_same<Z, double>::value> rotate(const JacobiRotationParameters& jacobi_rotation_parameters, const SpinComponent& component) {
+        this->spinor_basis[component].rotate(jacobi_rotation_parameters);
+    }
+
+    /**
+     *  Rotate the spinor basis for both the alpha- and beta components to another one using the unitary transformation matrix that corresponds to the given Jacobi rotation parameters
+     * 
+     *  @param jacobi_rotation_parameters       the Jacobi rotation parameters (p, q, angle) that are used to specify a Jacobi rotation: we use the (cos, sin, -sin, cos) definition for the Jacobi rotation matrix
+     * 
+     *  @note this function is only available for real spinor bases because Jacobi rotation parameters generate real rotations
+     */
+    template<typename Z = ExpansionScalar>
+    enable_if_t<std::is_same<Z, double>::value> rotate(const JacobiRotationParameters& jacobi_rotation_parameters) {
+        this->rotate(jacobi_rotation_parameters, SpinComponent::ALPHA);
+        this->rotate(jacobi_rotation_parameters, SpinComponent::BETA);
+    }
+
+    /**
+     *  Transform the spinor basis another one using the given transformation matrix
+     * 
+     *  @param component                the spin component
+     * 
+     *  @param T            the transformation matrix that transforms the requested component
+     */
+    void transform(const TransformationMatrix<ExpansionScalar>& T, const SpinComponent& component) {
+         this->spinor_basis[component].transform(T);
+    }
```

Comment:
Should we include some safeguard for the given dimensions of the transformation matrix?